pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,296 @@
+	// Linux contains configurations specific to Linux containers.
+	Linux *LinuxSandboxConfig
+}
+
+// LinuxSandboxConfig holds platform-specific configuraions for Linux
+// containers.
+type LinuxSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sanbox's namespaces.
+	NamespaceOptions NamespaceOptions
+	// SELinux is the SELinux context to be applied to all containers.
+	SELinux *api.SELinuxOptions
+}
+
+// NamespaceOptions provides options for Linux namespaces.
+type NamespaceOptions struct {
+	// HostNetwork uses the host's network namespace.
+	HostNetwork bool
+	// HostPID uses the host's pid namesapce.
+	HostPID bool
+	// HostIPC uses the host's ipc namespace.
+	HostIPC bool
+}
+
+// DNSOptions specifies the DNS servers and search domains.
+type DNSOptions struct {
+	// Servers is a list of DNS servers of the cluster.
+	Servers []string
+	// Searches is a list of DNS search domains of the cluster.
+	Searchs []string
+}
+
+type SandboxState string
+
+const (
+	// SandboxActive means the sandbox is functioning properly.
+	SandboxActive SandboxState = "active"
+	// SandboxInactive means the sandbox is not functioning properly.
+	SandboxInactive SandboxState = "inactive"
+)
+
+// SandboxFilter is used to filter a list of Sandboxes.
+type SandboxFilter struct {
+	// Name of the sandbox.
+	Name string
+	// ID of the sandbox.
+	ID string
+	// State of the sandbox.
+	State string
+	// Annotations of the sandbox. A container needs to have all listed
+	// annotations to be sandbox a match.
+	Annotations map[string]string
+}
+
+// SandboxListItem contains minimal information about a sandbox.
+type SandboxListItem struct {
+	ID    string
+	State SandboxState
+}
+
+// Sandbox provides isolation with resource requirements in which containers
+// can be run.
+type Sandbox struct {
+	// ID of the sandbox.
+	ID string
+	// Config of the sandbox used to create the sandbox.
+	Config SandboxConfig
+	// Status of the sandbox.
+	Status SandboxStatus
+}
+
+// SandboxStatus contains the status of the sandbox.
+type SandboxStatus struct {
+	// State of the sandbox.
+	State SandboxState
+	// Network contains network status if network is handled by the runtime.
+	Network *SandboxNetworkStatus
+	// Status specific to a Linux sandbox.
+	Linux *LinuxSandboxStatus
+}
+
+type SandboxNetworkStatus struct {
+	IP string
+}
+
+// Namespaces contains paths to the namespaces.
+type Namespaces struct {
+	// Network is the path to the network namespace.
+	Network string
+}
+
+// LinuxSandBoxStatus contains status specific to Linux sandboxes.
+type LinuxSandboxStatus struct {
+	// Namespaces contains paths to the sandbox's namespaces.
+	Namespaces *Namespaces
+}
+
+// Resources contains the CPU/memory resource requirements.
+type Resources struct {
+	// CPU resource requirement.
+	CPU api.ResourceRequirements
+	// Memory resource requirement.
+	Memory api.ResourceRequirements
```

Comment:
Even for pod level, we may not always be able to enforce these. What about windows?